core/pubnub_ccore_pubsub.c

```diff
@@ -115,22 +120,40 @@ char const* pbcc_get_channel(struct pbcc_context* pb)
 }
 
 
-void pbcc_set_uuid(struct pbcc_context* pb, const char* uuid)
+enum pubnub_res pbcc_set_uuid(struct pbcc_context* pb, const char* uuid)
 {
-    if (uuid != NULL) {
-        PUBNUB_ASSERT_OPT(strlen(uuid) < sizeof pb->uuid);
-        strncpy(pb->uuid, uuid, sizeof pb->uuid);
-        pb->uuid[(sizeof pb->uuid) - 1] = '\0';
+    if (pb->uuid_len > 0) {
+        pb->uuid_len = 0;
+        free(pb->uuid);
     }
-    else {
-        pb->uuid[0] = '\0';
+    pb->uuid = (char *)"";
+    if (uuid != NULL) {
+        int len = strlen(uuid);
```

Comment:
Should we skip allocation if `len == 0`?

Reply:
@pnehrer woooow, I missed this one. Thanks for noticing this one.

Reply:
Suggested changes has been added in 644daea.